cou/steps/hypervisor.py

```diff
@@ -29,23 +31,49 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class HypervisorMachine:
-    """Hypervisor machine containing units for multiple applications."""
-
-    name: str
-    units: list[COUUnit]
-
-
-@dataclass(frozen=True)
+@dataclass
 class HypervisorGroup:
     """Group of hypervisors.
 
     For example, this can represent a group of all hypervisors in a single availability zone.
     """
 
     name: str
-    hypervisors: list[HypervisorMachine]
+    app_units: dict[str, list[COUUnit]]
+
+    def __eq__(self, other: Any) -> bool:
+        """Equal magic method for HypervisorGroup.
+
+        :param other: HypervisorGroup object to compare.
+        :type other: Any
+        :return: True if equal False if different.
+        :rtype: bool
+        """
+        if not isinstance(other, HypervisorGroup):
+            return NotImplemented
+
+        return other.name == self.name
+
+
+class AZs(defaultdict):
+    """AZs dictionary object with default value HypervisorGroup."""
+
+    def __init__(self) -> None:
+        """Initialize the Hypervisor class.
+
+        The AZs represent default dict, with predefined HypervisorGroup as default value.
+        """
+        super().__init__(default_factory=None)
+
+    def __missing__(self, key: str) -> HypervisorGroup:
+        """Handle missing key in AZs.
+
+        If key is missing in the AZs dict, the default value
+        HypervisorGroup(name=key, apps=defaultdict(list) will be used. Like this we can always
+        access the az["my-az"].name == "my-az" or az["my-az"].apps["my-app"].
+        """
+        self[key] = HypervisorGroup(name=key, app_units=defaultdict(list))
+        return self[key]
 
 
 def verify_apps(apps: list[OpenStackApplication]) -> None:
```

Comment:
I guess we can remove this and add if necessary

Reply:
We are doing dev branch, so no point to remove it here, since it's not related. Resolving this comment